app/code/community/Reverb/ReverbSync/Helper/Orders/Update/Abstract.php

```diff
@@ -15,10 +18,71 @@
     const REASON_COMPLETE = 'The order has already been marked as complete';
     const REASON_CLOSED = 'The order has already been marked as closed';
     const REASON_DEFAULT = 'The order can not be cancelled';
-    
+
+    /**
+     * @return string
+     */
     abstract public function getUpdateAction();
 
     /**
+     * @var null|Reverb_ReverbSync_Helper_Orders_Creation_Address
+     */
+    protected $_orderAddressCreationHelper = null;
+
+    /**
+     * Returns the updated order shipping address for the $magentoOrder order if $orderArgumentObject contains
+     *      different address data values than those on the $magentoOrder object's shipping address
+     * Returns null otherwise
+     *
+     * @param Mage_Sales_Model_Order $magentoOrder
+     * @param stdClass $orderArgumentObject
+     * @return Mage_Sales_Model_Order_Address|null
+     */
+    public function updateAndReturnOrderShippingAddressIfNecessary(Mage_Sales_Model_Order $magentoOrder, stdClass $orderArgumentObject)
+    {
+        // Get the shipping address for the order
+        $orderShippingAddress = $magentoOrder->getShippingAddress();
+        // Get the shipping address defined in the $argumentObject
+        $argumentObjectShippingAddress = $orderArgumentObject->shipping_address;
+        $argumentsObjectOrderShippingAddress = $this->_getOrderAddressCreationHelper()
+                                                    ->getCustomerAddressForOrderByArgumentsObject($argumentObjectShippingAddress);
+        $argumentsObjectOrderShippingAddress->implodeStreetAddress();
+        // Check to see if the data points between the $orderShippingAddress and $argumentsObjectOrderShippingAddress
+        //      objects differ
+        $arguments_object_order_data_array = $argumentsObjectOrderShippingAddress->getData();
+        // Reverb will return abbreviated state values for the region. Magento will convert the region to the full
+        //  state name since getCustomerAddressForOrderByArgumentsObject() will return the region id for the address
+        //  As such, we don't want the region field to trigger an update, we will only trigger an update if the
+        //  region_id value is different
+        // We expect the 'region' index to be set on $arguments_object_order_data_array, but check isset() just in case
+        if (isset($arguments_object_order_data_array['region']))
+        {
+            unset($arguments_object_order_data_array['region']);
+        }
+
+        $has_a_field_been_updated = false;
+        foreach($arguments_object_order_data_array as $field => $arguments_object_value)
+        {
+
+            $order_address_value = $orderShippingAddress->getData($field);
+            if ($order_address_value != $arguments_object_value)
```

Comment:
do we need to trim whitespace from the values here? Is there any downside of saying an address has updated even if only leading/trailing whitespace is different?